src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,75 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const numberOfPages = Math.ceil(total / perPage);
+  const pages = Array.from({ length: numberOfPages }, (_, i) => i + 1);
+
+  const isFirstPageSelected = currentPage === 1;
+  const isLastPageSelected = currentPage === pages.length;
+
+  const pageChangeHandler = (page: number) => {
```

Comment:
Consider wrapping the pageChangeHandler function with useCallback to prevent unnecessary re-creations on every render.